kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KopBrokerLookupManager.java

```diff
@@ -122,21 +120,31 @@ private CompletableFuture<InetSocketAddress> lookupBroker(final String topic) {
 
     public CompletableFuture<Boolean> isTopicExists(final String topic) {
         CompletableFuture<Boolean> future = new CompletableFuture<>();
-        this.adminClient.topics().getPartitionedTopicMetadataAsync(topic)
-                .whenComplete((__, ex) -> {
+        TopicName topicName = TopicName.get(topic);
+        this.pulsar.getBrokerService().fetchPartitionedTopicMetadataAsync(TopicName.get(topic))
+                .whenComplete((metadata, ex) -> {
+                    if (metadata.partitions == 0) {
+                        internalCheckTopicExists(topicName).thenAccept(future::complete)
+                                .exceptionally(throwable -> {
+                                    log.error("Check topic exists has exception.", throwable);
+                                    future.complete(true);
```

Comment:
Do you intend to ignore the exception?

Reply:
Yes, if it has an exception, we should let the `TransactionMarkerChannelManager` retry.